jujugui/static/gui/src/app/store/env/bakery.js

```diff
@@ -18,721 +18,535 @@ with this program.  If not, see <http://www.gnu.org/licenses/>.
-        var savedMacaroon = this.getMacaroon();
-        if (savedMacaroon !== null) {
-          callback(null, savedMacaroon);
-          return;
-        }
-        if (!this.staticMacaroonPath) {
-          callback('Static macaroon path was not defined.');
-          return;
-        }
-        return this.webhandler.sendGetRequest(
-          this.staticMacaroonPath, null, null, null, false, null,
-          this._interactivePrefetch.bind(this, callback));
-      },
-
-      /**
-        Handler for the interactive macaroon prefetch.
-
-        @method _interactivePrefetch
-        @param {Function} callback The callback to be called after success
-          or failure.
-        @param {Object} The response object from the staticMacaroonPath fetch.
-        @return {undefined}
-      */
-      _interactivePrefetch: function(callback, res) {
-        var macaroon = {};
-        try {
-          macaroon = JSON.parse(res.target.responseText);
-        } catch(e) {
-          callback(e);
-          return;
-        }
-        this._authenticate(macaroon, function() {
-          callback(null, this.getMacaroon());
-        }.bind(this), callback);
-      },
-
-      /**
-       Prepare and return HTTP request headers for macaraq requests.
-
-       @param {Object} overrides Any additional headers.
-       @return {Object} The resulting request headers.
-      */
-      _prepareHeaders: function(overrides) {
-        const headers = {'Bakery-Protocol-Version': PROTOCOL_VERSION};
-        Object.keys(overrides || {}).forEach(key => {
-          headers[key] = overrides[key];
+    Serialize the given macaroons.
+
+    @param {Array} macaroons The macaroons to be serialized.
+    @return {string} The resulting serialized string.
+  */
+  const serialize = macaroons => {
+    return btoa(JSON.stringify(macaroons));
+  };
+
+  /**
+    De-serialize the given serialized macaroons.
+
+    @param {string} serialized The serialized macaroons.
+    @return {Array} The resulting macaroon slice.
+  */
+  const deserialize = serialized => {
+    return JSON.parse(atob(serialized));
+  };
+
+  /**
+    A macaroon bakery implementation.
+
+    The bakery implements the protocol used to acquire and discharge macaroons
+    over HTTP.
+  */
+  const Bakery = class Bakery {
+
+    /**
+      Initialize a bakery with the given HTTP client (used to make requests)
+      and storage (used to persist macaroons).
+
+      @param {Object} httpClient The client used to send HTTP requests. It must
+        implement a "send{Method}Request" method for every HTTP method. An
+        instance of WebHandler is usually provided.
+      @param {Object} storage The storage used to persist macaroons. It must
+        implement the following interface:
+          - get(key) -> value;
+          - set(key, value, callback): the callback is called without arguments
+            when the set operation has been performed;
+          - clear(): clear all stored data.
```

Comment:
We can remove clear from the storage contract as it is never used by the bakery itself, only by code external to the bakery. So the bakery storage can implement only set and get.